starknet_py/net/signer/stark_curve_signer.py

```diff
@@ -117,8 +118,20 @@ def _sign_deploy_account_transaction(self, transaction: DeployAccount) -> List[i
         r, s = message_signature(msg_hash=tx_hash, priv_key=self.private_key)
         return [r, s]
 
-    def sign_message(self, typed_data: TypedData, account_address: int) -> List[int]:
-        typed_data_dataclass = TypedDataDataclass.from_dict(data=typed_data)
+    def sign_message(
+        self, typed_data: Union[TypedData, TypedDataDataclass], account_address: int
+    ) -> List[int]:
+        if isinstance(typed_data, dict):
+            warnings.warn(
+                "TypedData as dict has been deprecated. Use starknet_py.utils.TypedData dataclass instead.",
```

Comment:
`TypedDataDataclass`, but it is a bad name. Can we name it differently if people should use it directly?

Reply:
There's no `TypedDataDataclass`, it's just used as an import alias so we can import both actual TypedData dataclass and a typeddict (used only for typing) some methods.

Reply:
Hmmm can't we leave it like it was? I think we can accept the dict as we did before, isn't it easier for users?
Also, what is displayed in docs for users? Will it be `Union[TypedData, TypedData]`? 🤔

Reply:
Well it is `Union[TypedData, TypedData]` 😅 so the argument about names makes more sense now. What renaming would you recommend?

About accepting dicts: for signing transactions we're already accepting a constructed tx object, so it seemed to me that'd make more sense to do the same for the typeddata: Using dicts is more for being user friendly in the higher level account interface (that abstracts using signer from you).

Reply:
If we want to switch to using typed data then we could type it as dict | TypedData 🤔

Reply:
I guess we could, we'll be loosing better typing from typeddict we now have, but perhaps we can accept that.

Reply:
Right, but users shouldn't be using the old way anyway